pkg/broker/container_manager.go

```diff
@@ -0,0 +1,25 @@
+package broker
+
+import (
+	"context"
+
+	"github.com/docker/docker/client"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+type ContainerConfig struct {
+	Name         string
+	Image        string
+	ExposedPorts []string
+}
+
+//go:generate counterfeiter -o fakes/fake_docker_client.go . DockerClient
+type DockerClient interface {
+	client.ImageAPIClient
+	client.ContainerAPIClient
+}
+
+//go:generate counterfeiter -o fakes/fake_container_manager.go . ContainerManager
+type ContainerManager interface {
+	Provision(ctx context.Context, cc *ContainerConfig) (brokerapi.ProvisionedServiceSpec, error)
```

Comment:
Are we not not attempting to implement some sort of GetRunningContainers as part of this? It's needed for both Deprovision and Bind, right? So that's going to be a conflict as both will have to attempt to implement that.

Or is there some way to do that that I'm not seeing?

Reply:
why is get running necessary?

Reply:
How am i supposed to tell the docker client to shut down running container foo? I have what, our internal id for the instance, no reference to the actual client because it's abstracted behind this interface, and the backend client shouldn't really care about our internal ID's and such, right? Same for Bind, we're going to have to talk at the specific instance, so we have to get where it's running somehow.

Reply:
Yes, I understand now. I think we missed adding a comment somewhere. The initial plan was to to pass through the instance_id as the name. That way there is a direct correspondence to the lower layer and it is all stateless for that purpose.

Reply:
Does that mean for the time being we aren't storing even in memory what we have running? I could have missed it in the rest of the PR

Reply:
I think we should add the in memory storage too

Reply:
Given that OSB brokers are stateless, I personally consider this an acceptable design decision permanently.
Docker requires container names to be unique.
Kubernetes requires pod/deployment/(anything at the same 'level', etc) to have unique names.